background.js

```diff
@@ -0,0 +1,292 @@
+(function() {
+  // semi-global object to hang methods on so they're available even when
+  // `this` has changed
+  var txgh = {};
+
+  // Default configuration options. Set config by using the little icon for
+  // this extension that appears to the right of the omnibox (i.e. URL bar).
+  var defaultConfig = {
+    // Your Github token to use when communicating with the Github API. I'd
+    // suggest creating a personal access token for use with this extension
+    // as opposed to an oauth token.
+    githubToken: null,
+
+    // A map of Transifex project slugs to Github repositories.
+    projectMap: {},
+
+    // A map of Transifex organizations to Github organizations.
+    organizationMap: {},
+
+    // List of regexes to identify links in the bodies of pull requests. These
+    // links will appear in the "Links" section in Transifex.
+    linkRegexes: [
+      "(https?:\\/\\/[\\w.]+atlassian\\.net\\/[\\/\\w-]+)"
+    ]
+  };
+
+  // basically main
+  chrome.runtime.onConnect.addListener(function(port) {
+    port.onMessage.addListener(function(message, port) {
+      switch(message.method) {
+        // triggered whenever the content script wants to refresh the page/tab
+        case 'updateTab':
+          txgh.triggerTabUpdate(port, port.sender.tab);
+          break;
+
+        // The response to the getResource message, sent from the content script.
+        // Since the content script is the only code that can access the DOM,
+        // the background script has to send the getResource message to pull
+        // the name of the branch off the page. The content script uses jQuery
+        // to grab the branch name (i.e. resource name), then hands it back to
+        // the background script via the getResourceResponse message.
+        case 'getResourceResponse':
+          var tab = port.sender.tab;
+
+          if (message.tabId !== tab.id) {
+            return;
+          }
+
+          txgh.getConfig().then(function(config) {
+            var routeInfo = txgh.getRouteInfo(tab.url, config);
+            var branch = txgh.branchFromResource(message.resource);
+            var pullRequest = txgh.findPullRequestForBranch(routeInfo, branch, config);
```

Comment:
Well, the `getResourceResponse` message won't be sent unless `routeInfo()` returns something other than `null`. The cycle goes like this:

1. `triggerTabUpdate` gets called and sends the `getResource` message. If `routeInfo()` is `null`, `getResource` isn't sent.
2. The content script receives `getResource` and responds by sending `getResourceResponse`.
3. The background script receives `getResourceResponse`, queries for the pull request, and updates the tab with the appropriate links.

So, if `routeInfo()` returns null this whole message sending chain gets halted. I can add another check for `null` on line 52 just in case... but it seems unnecessary.